sympy/printing/pretty/pretty.py

```diff
@@ -1902,27 +1902,26 @@ def _print_Mul(self, product):
             return prettyForm.__mul__(*a)/prettyForm.__mul__(*b)
 
     # A helper function for _print_Pow to print x**(1/n)
-    def _print_nth_root(self, base, expt):
+    def _print_nth_root(self, base, root):
         bpretty = self._print(base)
 
         # In very simple cases, use a single-char root sign
         if (self._settings['use_unicode_sqrt_char'] and self._use_unicode
-            and expt is S.Half and bpretty.height() == 1
+            and root == 2 and bpretty.height() == 1
             and (bpretty.width() == 1
                  or (base.is_Integer and base.is_nonnegative))):
             return prettyForm(*bpretty.left('\N{SQUARE ROOT}'))
 
         # Construct root sign, start with the \/ shape
         _zZ = xobj('/', 1)
         rootsign = xobj('\\', 1) + _zZ
-        # Make exponent number to put above it
-        if isinstance(expt, Rational):
-            exp = str(expt.q)
-            if exp == '2':
-                exp = ''
-        else:
-            exp = str(expt.args[0])
-        exp = exp.ljust(2)
+        # Constructing the number to put on root
+        rpretty = self._print(root)
+        # If n is not an integer or simple symbol, print without root sign
+        if not root.is_Integer and (rpretty.height() > 1 or rpretty.width() > 1):
+            return self._print(base)**self._print(1/root)
```

Comment:
What changes if we use the simpler condition?
```suggestion
        # roots look bad if they are not a single line
        if rpretty.height() != 1:
            return self._print(base)**self._print(1/root)
```

Reply:
Really the only change is that
```
z = Symbol("x_17")
pprint(7**(1/z))
```
changes from
```
   1 
 ───
 x₁₇
7  
```
to 
```
x₁₇___
 ╲╱ 7 
```